core/modules/wdb/QueryRunner.cc

```diff
@@ -435,6 +443,7 @@ bool QueryRunner::_dispatchChannel() {
     // Collect the result in _transmitData. When a reasonable amount of data has been collected,
     // or there are no more rows to collect, pass _transmitData to _sendChannel.
     try {
+        _initTransmit(); // set _transmit
```

Comment:
Is this a bug fix? Why it wasn't here before, and what effect it will cause? It's not mentioned in the description for the ticket.

Reply:
It's a bug fix for a problem that only occurs when a query is cancelled and its very hard to reproduce on a local installation, but the cluster notices right away. If the query was cancelled, _initTransmit() wouldn't be called, and there's a segv (this killed a lot of workers on the cluster) when this function tries to build an error message. Making sure _transmit is valid prevents the crash.

Reply:
Also, before this ticket, _cancelled was never true, so this crash would never happen.